### DISCUSSION TITLE
TrakSTAR_ReferenceStylus-Capture.xml

### DISCUSSION THREAD

**Discussion by whucsu** [PlusToolkit/PlusLib#974] [Q&A]
I get a device : NDI trakSTAR, and I want to connect it with PLUS, and then visualize the data in Slicer.
And the tutorials on SlicerIGT(http://www.slicerigt.org/wp/user-tutorial/) gives some help.
Here, the Slicer UI means  that it can communicate with the device, but I cannot get the position data.
The  default config file PlusDeviceSet_Server_Ascension3DG.xml is used in the PLUS server, other early tutorial used the  TrakSTAR_ReferenceStylus-Capture.xml , but it can not be found anywhere.
![e4663154ccfccadddb8ec352bda1447](https://user-images.githubusercontent.com/92866343/183824571-ed09b43e-2521-4255-9e14-98f39b74b897.jpg)
Not the transorm as needed.
![76fadb9fd1c4d7fcb53db792490c81f](https://user-images.githubusercontent.com/92866343/183824631-025ab94a-216e-47e7-b90c-0f83deb977ea.jpg)

**Comment by Sunderlandkyl** [comment]
Can you attach your log file from Plus?

**Comment by whucsu** [comment]
thanks for your kind heart.
it seem that no buffer data,  especially using fCal. here is the log file
[081122_163454_PlusLog.txt](https://github.com/PlusToolkit/PlusLib/files/9313037/081122_163454_PlusLog.txt)
.

**Reply by Sunderlandkyl** [reply]
What devices do you have connected to your trakSTAR?

The log indicates that no sensors are plugged into port 1 (Reference sensor) or 2 (Stylus sensor).

**Reply by whucsu** [reply]
only the deivice, and one sensor, no other device.
![image](https://user-images.githubusercontent.com/92866343/184284246-78ba2c88-a007-49a4-be22-5b26e2745b6e.png)

**Reply by whucsu** [reply]
> The log indicates that no sensors are plugged into port 1 (Reference sensor) or 2 (Stylus sensor).
the device and the sensor from the NDI , no others..
And the device works well using NDI's drivers and utility application.

**Reply by Sunderlandkyl** [reply]
According to the log, the config file specifies that 3 sensors should be connected (from left-to-right):
- Probe sensor
- Reference sensor
- Stylus sensor

You need to remove the sensors from the config file that you don't plan to use.

**Reply by whucsu** [reply]
Buffer item is not in the buffer
Unable to get timestamp from ReferenceToTracker
what does this two problems mean?  I just need the position information, only one sensor.

**Reply by whucsu** [reply]
I change the config file as follows:
```
<PlusConfiguration version="2.3">
  <DataCollection StartupDelaySec="1.0" >
    <DeviceSet 
      Name="PlusServer: Ascension 3DG tracker"
      Description="Broadcasting tool tracking data through OpenIGTLink
Ascension3DG sensors should be plugged in to the Ascension3DG DriveBay mounted on Ultrasonix US in the following order from to leftmost slot (Transducer 1) to the right: 1 Probe, 2 Reference, 3 Stylus." 
    />
    <Device
      Id="TrackerDevice" 
      Type="Ascension3DG" 
      FilterAcWideNotch="1"
      ToolReferenceFrame="Tracker" >
      <DataSources>
        <DataSource Type="Tool" Id="Probe" PortName="0"  BufferSize="500" AveragedItemsForFiltering="20" />
      </DataSources>
      <OutputChannels>
        <OutputChannel Id="TrackerStream" >
          <DataSource Id="Probe"/>
        </OutputChannel>
      </OutputChannels>
    </Device>
  </DataCollection>

  <PlusOpenIGTLinkServer 
    MaxNumberOfIgtlMessagesToSend="1" 
    MaxTimeSpentWithProcessingMs="50" 
    ListeningPort="18944" 
    OutputChannelId="TrackerStream" > 
    <DefaultClientInfo> 
      <MessageTypes> 
        <Message Type="TRANSFORM" />
      </MessageTypes>
      <TransformNames> 
        <Transform Name="ProbeToTracker" />
      </TransformNames>
    </DefaultClientInfo>
  </PlusOpenIGTLinkServer>
</PlusConfiguration>
```

The result is little better, but still no data.
The log file is :
time|level|timeoffset|message|location
081422_112120.786|INFO|000.002000| Software version: Plus-2.8.0.62873a16 - Win32| in vtkPlusLogger(188)
081422_112120.786|INFO|000.002000| System start timestamp: 156860| in E:\D\PSNPb\PlusLib\src\PlusCommon\vtkPlusAccurateTimer.cxx(51)
081422_112120.850|INFO|000.067000|  Software version: Plus-2.8.0.62873a16 - Win32| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(122)
081422_112120.850|INFO|000.067000| Logging at level 3 to file: C:/PlusApp-2.8/data/081422_112120_PlusLog.txt| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(123)
081422_112121.283|INFO|000.500000| Supported devices: 
  - 3dConnexion (ver: Plus-2.8.0)
  - Ascension3DG (ver: Plus-2.8.0)
  - AuroraTracker (ver: NDICAPI-1.7)
  - BkProFocusOem (ver: Plus-2.8.0)
  - BrachyTracker (ver: Plus-2.8.0)
  - CertusTracker (ver: Optotrak API (Win32 DLL Version) Version 3.01.03)
  - ChRobotics (ver: Plus-2.8.0)
  - Epiphan (ver: Plus-2.8.0)
  - FakeTracker (ver: Plus-2.8.0)
  - GenericSerialDevice (ver: Plus-2.8.0)
  - ICCapturing (ver: The Imaging Source UDSHL-3.4)
  - ImageProcessor (ver: Plus-2.8.0)
  - IntelRealSense (ver: Plus-2.8.0)
  - Microchip (ver: Plus-2.8.0)
  - MicronTracker (ver: MTC-3.7.6.8)
  - MmfVideo (ver: Plus-2.8.0)
  - NDITracker (ver: NDICAPI-1.7)
  - NoiseVideo (ver: Plus-2.8.0)
  - OpenCVVideo (ver: Plus-2.8.0)
  - OpenIGTLinkTracker (ver: OpenIGTLink v3.1.0)
  - OpenIGTLinkVideo (ver: OpenIGTLink v3.1.0)
  - OptiTrack (ver: Plus-2.8.0)
  - OpticalMarkerTracker (ver: Plus-2.8.0)
  - PhidgetSpatial (ver: Plus-2.8.0)
  - PolarisTracker (ver: NDICAPI-1.7)
  - SavedDataSource (ver: Plus-2.8.0)
  - SonixPortaVideo (ver: UltrasonixSDK-6.1.1)
  - SonixVideo (ver: UltrasonixSDK-6.1.1)
  - USDigitalEncodersTracker (ver: Plus-2.8.0)
  - UsSimulator (ver: Plus-2.8.0)
  - VFWVideo (ver: Plus-2.8.0)
  - VirtualBufferedCapture (ver: Plus-2.8.0)
  - VirtualCapture (ver: Plus-2.8.0)
  - VirtualDiscCapture (ver: Plus-2.8.0)
  - VirtualMixer (ver: Plus-2.8.0)
  - VirtualSwitcher (ver: Plus-2.8.0)
  - VirtualTextRecognizer (ver: Plus-2.8.0)
  - VirtualVolumeReconstructor (ver: Plus-2.8.0)
  - WitMotionTracker (ver: Plus-2.8.0)
| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(127)
081422_112121.291|INFO|000.508000| Server host name: LAPTOP-VACMMNS3| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(158)
081422_112121.307|INFO|000.524000| Server IP addresses: 169.254.239.169, 169.254.118.235, 169.254.212.25, 192.168.232.1, 192.168.200.1, 192.168.1.8, 127.0.0.1| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(178)
081422_112121.307|INFO|000.524000| Start remote control server at port: 18904| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(187)
081422_112218.557|INFO|057.774000| Connect using configuration file: C:\PlusApp-2.8\config\Test-3DG.xml| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(639)
081422_112218.559|INFO|057.775000| Server process command line: "C:/PlusApp-2.8/bin/PlusServer.exe" --config-file="Test-3DG.xml" --verbose=3| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(320)
081422_112219.266|INFO|058.483000| Server process started successfully| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(329)
081422_112223.707|INFO|062.924000|SERVER> System start timestamp: 156923
081422_112223.708|INFO|062.925000|SERVER> Software version: Plus-2.8.0.62873a16 - Win32
081422_112223.710|INFO|062.926000|SERVER> Logging at level 3 (INFO) to file: C:/PlusApp-2.8/data/081422_112223_PlusLog.txt
081422_112223.714|INFO|062.931000|SERVER> Server status: Reading configuration.
081422_112223.715|INFO|062.932000|SERVER> Server status: Connecting to devices.
081422_112234.171|INFO|073.388000|SERVER> Server status: Starting servers.
081422_112234.192|INFO|073.409000|SERVER> Plus OpenIGTLink server listening on IPs: 169.254.239.169, 169.254.118.235, 169.254.212.25, 192.168.232.1, 192.168.200.1, 192.168.1.8, 127.0.0.1 -- port 18944
081422_112234.234|INFO|073.451000|SERVER> Server status: Server(s) are running.
081422_112234.237|INFO|073.454000|SERVER> Press Ctrl-C to quit.
081422_112348.198|INFO|147.415000|SERVER> Received new client connection (client 1 at 127.0.0.1:18944). Number of connected clients: 1
081422_112348.230|INFO|147.446000|SERVER> OpenIGTLink broadcasting started. No data was available between 0-74.5951sec, therefore no data were broadcasted during this time period.
081422_112555.090|INFO|274.307000|SERVER> Received new client connection (client 2 at 192.168.200.1:18944). Number of connected clients: 2
081422_112754.222|INFO|393.439000| Server process stop request sent successfully| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(512)
081422_112755.886|INFO|395.103000| Server process stopped successfully| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(531)
081422_112755.888|INFO|395.105000| Disconnect request successful| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(632)

Still Need for help.

**Reply by Sunderlandkyl** [reply]
Can you try removing these attributes from the probe tool element?
> BufferSize="500" AveragedItemsForFiltering="20"
